packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -0,0 +1,1144 @@
+/// [NavigationBarDestination]s).
+///
+/// Usage:
+/// ```dart
+/// Scaffold(
+///   bottomNavigationBar: NavigationBar(
+///     onTap: (i) => setState(() => _currentPageIndex = i),
+///     selectedIndex: _currentPageIndex,
+///     destinations: [
+///       NavigationBarDestination(
+///         icon: Icon(Icons.explore),
+///         label: 'Explore',
+///       ),
+///       NavigationBarDestination(
+///         icon: Icon(Icons.commute),
+///         label: 'Commute',
+///       ),
+///       NavigationBarDestination(
+///         selectedIcon: Icon(Icons.bookmark),
+///         icon: Icon(Icons.bookmark_border),
+///         label: 'Saved',
+///       ),
+///     ],
+///   ),
+/// ),
+/// ```
+class NavigationBar extends StatelessWidget {
+  /// Creates a Material 3 Navigation Bar component.
+  const NavigationBar({
+    Key? key,
+    this.animationDuration = const Duration(milliseconds: 500),
+    this.selectedIndex = 0,
+    required this.destinations,
+    this.onDestinationSelected,
+    this.elevation = 8,
+    this.backgroundColor,
+    this.height,
+    this.labelBehavior = NavigationBarDestinationLabelBehavior.alwaysShow,
+  }) : super(key: key);
+
+  /// Determines how long the animation is for each destination as it goes from
+  /// unselected to selected.
+  final Duration animationDuration;
+
+  /// Determines which destination from [destinations] is currently selected.
+  ///
+  /// When this is updated, the destination (from [destinations]) at
+  /// [selectedIndex] goes from unselected to selected. It will animate its
+  /// [NavigationBarDestinationInfo.selectedAnimation] from 0.0 to 1.0.
+  final int selectedIndex;
+
+  /// The list of destinations (usually [NavigationBarDestination]s) in this
+  /// [NavigationBar].
+  ///
+  /// When [selectedIndex] is updated, the destination from this list at
+  /// [selectedIndex] will animate from 0 (unselected) to 1.0 (selected). When
+  /// the animation is increasing or completed, the destination is considered
+  /// selected, when the animation is decreasing or dismissed, the destination
+  /// is considered unselected.
+  ///
+  /// Any widget in this list will have access to a
+  /// [NavigationBarDestinationInfo] inherited widget.
+  final List<Widget> destinations;
+
+  /// Called when one of the [destinations] is selected.
+  ///
+  /// This callback usually updates the int passed to [selectedIndex].
+  final ValueChanged<int>? onDestinationSelected;
+
+  /// The z-coordinate of this [NavigationBar].
+  final double elevation;
+
+  /// The color of the [NavigationBar] itself.
+  ///
+  /// If null, defaults to [ColorScheme.surface].
+  final Color? backgroundColor;
+
+  /// The height of the [NavigationBar]'s Material, from the bottom of the
+  /// screen to the top border of the widget, not including safe area padding
+  /// for things like system navigation.
+  ///
+  /// Defaults to 74 when labels are used (when [labelBehavior] is
```

Comment:
This comment is outdated now, It defaults to 80 all the time regardless of labelBehavior